### PR TITLE
Not null data in delivery address

### PR DESCRIPTION
**Pull request by sspooky13**
| Q             | A
| ------------- | ---
|Description, reason for the PR| On project has happened that country is not filled but [twig template expected country is always filled](https://github.com/shopsys/shopsys/blob/master/packages/framework/src/Resources/views/Admin/Form/deliveryAddressListFields.html.twig#L31) but entity (database) claims something else. I checked out and tested scenarios where you fill delivery address in order (PersonalInfoFormType) and in customer section (DeliveryAddressFormType) and always firstName, lastName, street, city, postcode, and country can't be null so I recommend these fields set as required or change template with delivery address data in country element as nullable: `deliveryAddress.country is not null ? deliveryAddress.country.name : ''` P.S.: I don't know how to solve  database where columns were nullable and now can't be nullable, I'll leave it up to you
|New feature| No
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by sspooky13** [comment]
I would like to add that I don't have SSFW, so I didn't run the tests, standards, or generated commands, sorry

File: packages/framework/src/Model/Customer/DeliveryAddress.php [hunk 1]
```diff
@@ -126,41 +126,41 @@ public function getCompanyName()
...
     {
         return $this->lastName;
     }
 
     /**
-     * @return string|null
+     * @return string
      */
-    public function getStreet()
+    public function getStreet(): string
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Customer/DeliveryAddress.php]: return typehints on getters for ORM annotated fields are forbidden in favor of better extensibility. 
Check phpstan to see issues, please

File: UPGRADE-15.0.md [hunk 2]
```diff
@@ -68,6 +68,7 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
     -   see #project-base-diff to update your project
 -   fix display advert in categories ([#3040](https://github.com/shopsys/shopsys/pull/3040))
     -   see #project-base-diff to update your project
+-   Fixed javascript validation company data in customer and first name, last name, street, city, postcode, and country can not be nullable in delivery address ([#2494](https://github.com/shopsys/shopsys/pull/2494))
```
Comment (grossmannmartin) [inline UPGRADE-15.0.md]:
```suggestion
#### make delivery address fields not nullable ([#2494](https://github.com/shopsys/shopsys/pull/2494))

-   class `Shopsys\FrameworkBundle\Form\Admin\Customer\DeliveryAddressFormType` was removed
-   check migration `Version20220725104726` if suits your needs
-   following fields in `Shopsys\FrameworkBundle\Model\Customer\DeliveryAddress` are no longer nullable:
    -   `$firstName`
    -   `$lastName`
    -   `$street`
    -   `$city`
    -   `$postcode`
    -   `$country`
-   see #project-base-diff to update your project
```

File: packages/framework/src/Form/Admin/Customer/DeliveryAddressFormType.php [hunk 3]
```diff
@@ -1,188 +0,0 @@
...
-use Symfony\Component\Form\AbstractType;
-use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
-use Symfony\Component\Form\Extension\Core\Type\FormType;
-use Symfony\Component\Form\Extension\Core\Type\TextType;
-use Symfony\Component\Form\FormBuilderInterface;
-use Symfony\Component\Form\FormInterface;
-use Symfony\Component\OptionsResolver\OptionsResolver;
-use Symfony\Component\Validator\Constraints;
-
-class DeliveryAddressFormType extends AbstractType
```
Comment (grossmannmartin) [inline packages/framework/src/Form/Admin/Customer/DeliveryAddressFormType.php:20]: `project-base/app/src/Form/Admin/Customer/DeliveryAddressFormTypeExtension.php` should be removed too, along with the reference in `project-base/app/config/forms.yaml`

File: packages/framework/src/Migrations/Version20220725104726.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+use Shopsys\MigrationBundle\Component\Doctrine\Migrations\AbstractMigration;
+
+class Version20220725104726 extends AbstractMigration
+{
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $this->sql('ALTER TABLE delivery_addresses ALTER first_name SET NOT NULL');
```
Comment (grossmannmartin) [inline packages/framework/src/Migrations/Version20220725104726.php:18]: We should handle existing values for all fields. 
I suggest to set an empty string like this 

```
$this->sql('UPDATE delivery_addresses SET first_name = \'\' WHERE first_name IS NULL');
$this->sql('ALTER TABLE delivery_addresses ALTER first_name SET NOT NULL');
```

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: packages/framework/src/Model/Customer/DeliveryAddress.php (hunk 1 above)
Comment (sspooky13) [reply packages/framework/src/Model/Customer/DeliveryAddress.php]: I did not notice, sorry. Fixed https://github.com/shopsys/shopsys/pull/2494/commits/6e6c837fc1e9a5871ce2038992c83e97828d094f

File: UPGRADE-15.0.md (hunk 2 above)
Comment (sspooky13) [reply UPGRADE-15.0.md]: fixed https://github.com/shopsys/shopsys/pull/2494/commits/ec3152d85903cc10b899c1672b8754afc2ba1de7

File: packages/framework/src/Form/Admin/Customer/DeliveryAddressFormType.php (hunk 3 above)
Comment (sspooky13) [reply packages/framework/src/Form/Admin/Customer/DeliveryAddressFormType.php:20]: true, fixed https://github.com/shopsys/shopsys/pull/2494/commits/73b18de19f53e2cac538b2ab7e68f5e01dc7b2bc

File: packages/framework/src/Migrations/Version20220725104726.php (hunk 4 above)
Comment (sspooky13) [reply packages/framework/src/Migrations/Version20220725104726.php:18]: I edit migration by your suggestion but I am not sure about `country_id`. I added random id but i am not sure if it's correct solution. Check it again please https://github.com/shopsys/shopsys/pull/2494/commits/45ffcba392b9f930e4e53af6d923e954d535ae9c

Comment (grossmannmartin) [reply packages/framework/src/Migrations/Version20220725104726.php:18]: 🤔  hmm that's quite a brain-teaser.

What about something like 

```
UPDATE delivery_address
SET country_id = (
    SELECT cd.country_id
    FROM customers c
    JOIN country_domains cd ON c.domain_id = cd.domain_id
    WHERE c.id = delivery_address.customer_id
      AND cd.enabled = TRUE
    ORDER BY cd.country_id ASC
    LIMIT 1
)
WHERE country_id IS NULL;
```

it's still arbitrary country, but at least country that's enabled on the customer's domain 🤷 (I did not test the migration, so there may be some mistake)

What do you think?

Comment (sspooky13) [reply packages/framework/src/Migrations/Version20220725104726.php:18]: Selecting from countries only that are allowed in domain is good idea 👌 It's still not 100% solution but I think it is good now. Changed in [commit](https://github.com/shopsys/shopsys/pull/2494/commits/98d55d13c19744e6753a27a6f160b82ddc47d500) and tested

**Review by grossmannmartin — APPROVED** [review]

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**